tests/test_unified_install_cache.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+import os
+import subprocess
+from subprocess import CalledProcessError
+from textwrap import dedent
+
+import pytest
+
+from pex import pip, resolver
+from pex.common import safe_rmtree
+from pex.pex_builder import PEXBuilder
+from pex.pex_info import PexInfo
+
+
+@pytest.fixture(scope='module')
+def repo_root():
+  return subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode('utf-8').strip()
+
+
+def test_issues_789_demo(repo_root, tmpdir):
+  # Imagine we've pre-resolved the requirements needed in our wheel house.
+  requirements = ['isort', 'setuptools']  # N.B.: isort doesn't declare its setuptools dependency.
+
+  wheelhouse = os.path.join(tmpdir, 'wheelhouse')
+  pip.spawn_download_distributions(
+    download_dir=wheelhouse,
+    requirements=requirements
+  ).wait()
+
+  # Also imagine this configuration is passed to a tool (PEX or a wrapper as in this test example)
+  # via the CLI.
+  configuration = dict(
+    requirements=requirements,
+    entry_point='isort.main:main',
+    resolver_settings=dict(
+      indexes=[],  # Turn off pypi.
+      find_links=[wheelhouse],  # Use our wheel house.
+      build=False,  # Use only pre-built wheels.
```

Comment:
To be clear, these configuration settings have nothing to do with ptex. This is me simulating a typical use case in a shop that wants to lock things down. The resolver settings could be left out to get a standard pypi resolve that's allowed to build wheels from sdists for example. Ptex would still use the resulting fully pinned resolve set and specify `--no-transitive`.